docs/source/app-dev/bindings-x-lang/index.rst

```diff
@@ -0,0 +1,46 @@
+.. Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Custom bindings
+###############
+
+This page gets you started with creating custom bindings for the Digital Asset distributed ledger.
+
+Introduction
+============
+
+Digital Asset currently provides bindings for the following programming languages:
+
+- Java
+
+- Scala
+
+- Node.js
+
+However it is possible to create bindings for any programming language supported by `gRPC <https://grpc.io/docs/>`_.
+
+What exactly is "bindings"? X language bindings consists of two main components:
+
+- Ledger API
+    Client "stubs" for X language, -- the remote ledger API that allows sending ledger commands and receiving ledger transactions back. Generated from gRPC protobuf definitions: https://github.com/digital-asset/daml/tree/master/ledger-api/grpc-definitions and documented on this page: :doc:`The Ledger API using gRPC </app-dev/grpc/index>`
+
+- Codegen
+    Code generator. A program that generates classes representing DAML contract templates in X language. These classes incorporate all boilerplate code responsible for constructing: (:ref:`com.digitalasset.ledger.api.v1.CreateCommand`) and (:ref:`com.digitalasset.ledger.api.v1.ExerciseCommand`) corresponding for each DAML contract template.
+
+Technically codegen is optional. It is possible to construct (:ref:`com.digitalasset.ledger.api.v1.CreateCommand`) and (:ref:`com.digitalasset.ledger.api.v1.ExerciseCommand`) manually. However this is very tedious and error-prone.
+
+Here is an example of constructing a :ref:`com.digitalasset.ledger.api.v1.CreateCommand` for contract template with one field: `party` ::
+
+
+  val fields = Seq(RecordField("party", Some(Value(party))))
+  val command =
+    Command.Create(
+      CreateCommand(
+        templateId = Some(templateId),
+        createArguments = Some(Record(Some(templateId), fields))))
+
+if you do not specify `party` field or type it's name or value incorrectly, the above code will compile but fail in run-time because you did not structure you create command correctly. Codegen is expected to simplify the command construction by providing an auto-generated utilities to help you constructing commands. E.g. here is how a similar contract instantiation would look if you are using a scala codegen to generate contract classes::
```

Comment:
```suggestion
If you do not specify `party` field or type its name or value incorrectly, or don't order the fields exactly as they are in the DAML template, the above code will compile but fail at run-time because you did not structure your create command correctly. Codegen is expected to simplify the command construction by providing auto-generated utilities to help you construct commands. Here is how a similar contract instantiation would look if you are using Scala codegen to generate contract classes::
```